File: number_compliment/number_compliment.rb
# @param {Integer} num
# @return {Integer}
def find_complement(num)
	num = num.to_s(2).split('').map! do |bit|
		if bit == '1'
			'0'
		else
			'1'
		end
	end
	num.join('').to_i(2)
end

p find_complement(5)

File: number_compliment/number_compliment.java
public class Solution {
    public int findComplement(int num) {
      String numBin =  Integer.toBinaryString(num);   
      char[] numBinArr = numBin.toCharArray();
      for(int i = 0;i < numBinArr.length;i++){
        if(numBinArr[i] == '1'){
          numBinArr[i] = '0';
        }else{
          numBinArr[i] = '1';
        }
      }
      return Integer.parseInt(new String(numBinArr), 2);
    }
}